### PR TITLE
feat(test): Add filebased config support for integration tests

### PR DESCRIPTION
**Pull request by Deniallugo**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

File: core/tests/ts-integration/src/env.ts [hunk 1]
```diff
@@ -45,25 +42,142 @@ export async function waitForServer() {
...
+    if (network.toLowerCase() == 'localhost') {
+        const testConfigPath = path.join(process.env.ZKSYNC_HOME!, `etc/test_config/constant`);
+        const ethTestConfig = JSON.parse(fs.readFileSync(`${testConfigPath}/eth.json`, { encoding: 'utf-8' }));
+        return ethers.Wallet.fromMnemonic(ethTestConfig.test_mnemonic as string, "m/44'/60'/0'/0/0").privateKey;
+    } else {
+        return ensureVariable(process.env.MASTER_WALLET_PK, 'Main wallet private key');
+    }
+}
+
+async function loadTestEnvironmentFromFile(chain: string): Promise<TestEnvironment> {
```
Comment (Deniallugo) [inline core/tests/ts-integration/src/env.ts:55]: This method has a bunch of similarities with  `loadTestEnvironmentFromEnv`. But soon we will remove `loadTestEnvironmentFromEnv` and keep the only option

**Review by aon — APPROVED** [review]

File: core/tests/ts-integration/src/env.ts [hunk 2]
```diff
@@ -45,25 +42,146 @@ export async function waitForServer() {
...
+        return ethers.Wallet.fromMnemonic(ethTestConfig.test_mnemonic as string, "m/44'/60'/0'/0/0").privateKey;
+    } else {
+        return ensureVariable(process.env.MASTER_WALLET_PK, 'Main wallet private key');
+    }
+}
+
+/*
+    Loads the environment for file based configs.
+ */
+
```
Comment (popzxc) [inline core/tests/ts-integration/src/env.ts:58]: I was grumpy about you forgetting empty lines, so you took the feedback and doubled down 😅

**Review by popzxc — DISMISSED** [review]
Approving not to block the merge in case it's urgent, but I think this PR could get a bit more love.

File: core/tests/ts-integration/src/env.ts [hunk 3]
```diff
@@ -195,3 +321,56 @@ function getTokens(pathToHome: string, network: string): L1Token[] {
...
+        fs.readFileSync(configPath, {
+            encoding: 'utf-8'
+        }),
+        {
+            customTags
+        }
+    );
+}
+
+function loadEcosystem(pathToHome: string): any {
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply core/tests/ts-integration/src/env.ts:340]: I think when we remove the env variables options, it won't be so messy

File: core/tests/ts-integration/src/env.ts [hunk 4]
```diff
@@ -45,25 +42,146 @@ export async function waitForServer() {
...
+
+    const baseTokenAddressL2 = L2_BASE_TOKEN_ADDRESS;
+    const l2ChainId = parseInt(genesisConfig.l2_chain_id);
+    const l1BatchCommitDataGeneratorMode = genesisConfig.l1_batch_commit_data_generator_mode as DataAvailabityMode;
+    let minimalL2GasPrice = generalConfig.state_keeper.minimal_l2_gas_price;
+    // TODO add support for en
+    let nodeMode = NodeMode.Main;
+
+    const validationComputationalGasLimit = parseInt(generalConfig.state_keeper.validation_computational_gas_limit);
+    // TODO set it properly
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply core/tests/ts-integration/src/env.ts:118]: together with en

**Review by aon — APPROVED** [review]

**Review by aon — APPROVED** [review]

**Pull request merged**